lib/src/tar/import.rs

```diff
@@ -139,6 +146,46 @@ fn parse_checksum(parent: &str, name: &Utf8Path) -> Result<String> {
     Ok(checksum)
 }
 
+// Parse a `.file-xattrs-link` symlink target into the corresponding checksum.
+fn parse_xattrs_link_target(path: &Utf8Path) -> Result<String> {
+    // Discard the relative parent.
+    let path = path.strip_prefix("..")?;
+
+    // Split the sharded checksum directory.
+    let parentname = path
+        .parent()
+        .map(|p| p.file_name())
+        .flatten()
+        .ok_or_else(|| anyhow!("Invalid path (no parent) {}", path))?;
+    if parentname.len() != 2 {
+        return Err(anyhow!("Invalid checksum parent {}", parentname));
+    }
+
+    // Split the filename (basename + objtype).
+    let fname = path
+        .file_name()
+        .map(Utf8Path::new)
+        .ok_or_else(|| anyhow!("Invalid filename {}", path))?;
+
+    // Ensure the symlink points to the correct object type.
+    let objtype = fname
+        .extension()
+        .ok_or_else(|| anyhow!("Invalid path (extension) {}", path))?;
+    if objtype != "file-xattrs" {
+        bail!("Invalid objpath {} for {}", objtype, path);
+    }
+
+    // Reassemble the target checksum and validate it.
+    let target = {
+        let basename = fname.as_str().trim_end_matches(".file-xattrs");
+        let checksum = format!("{}{}", parentname, basename);
+        validate_sha256(&checksum)?;
+        checksum
+    };
```

Comment:
I don't quite get the value of the nested scope here.  Hmm, would it be cleaner if e.g. `validate_sha256()` took ownership and returned it?  Then it could be:
`Ok(validate_sha256(format!("{}{}", parentname, basename)))`
or so.

Reply:
It just mirrors my mental logic organization, as the matching comment before that suggests. I removed this nesting to avoid doubts.

I also updated the validation function to transfer ownership as suggested.